packages/wallet/src/redux/protocols/new-ledger-funding/reducer.ts

```diff
@@ -1,42 +1,385 @@
+): ReturnVal {
+  if (
+    isAdvanceChannelAction(action) &&
+    action.protocolLocator === NEW_LEDGER_FUNDING_PROTOCOL_LOCATOR
+  ) {
+    const advanceChannelResult = advanceChannelReducer(
+      protocolState.postFundSetupState,
+      sharedData,
+      action,
+    );
+    sharedData = advanceChannelResult.sharedData;
+    return {
+      protocolState: {
+        ...protocolState,
+        postFundSetupState: advanceChannelResult.protocolState,
+      },
+      sharedData,
+    };
+  }
+  const existingDirectFundingState = protocolState.directFundingState;
+  const protocolStateWithSharedData = directFundingStateReducer(
+    existingDirectFundingState,
+    sharedData,
+    action,
+  );
+  const newDirectFundingState = protocolStateWithSharedData.protocolState;
+  const newProtocolState = { ...protocolState, directFundingState: newDirectFundingState };
+  sharedData = protocolStateWithSharedData.sharedData;
+
+  if (!isTerminal(newDirectFundingState)) {
+    return { protocolState: newProtocolState, sharedData };
+  }
+  if (isFailure(newDirectFundingState)) {
+    return { protocolState: failure({}), sharedData };
+  }
+  if (isSuccess(newDirectFundingState)) {
+    const channel = getChannel(sharedData.channelStore, newProtocolState.ledgerId);
+    if (!channel) {
+      throw new Error(`Could not find channel for id ${newProtocolState.ledgerId}`);
+    }
+    const { processId, ledgerId, channelId } = protocolState;
+    const advanceChannelResult = advanceChannelReducer(
+      protocolState.postFundSetupState,
+      sharedData,
+      advanceChannelClearedToSend({
+        processId,
+        protocolLocator: NEW_LEDGER_FUNDING_PROTOCOL_LOCATOR,
+      }),
+    );
+    sharedData = advanceChannelResult.sharedData;
+    const latestCommitment = getLatestCommitment(ledgerId, sharedData);
+    const proposedAllocation = [latestCommitment.allocation.reduce(addHex)];
+    const proposedDestination = [channelId];
+    const consensusUpdateResult = initializeConsensusUpdate(
+      processId,
+      ledgerId,
+      false,
+      proposedAllocation,
+      proposedDestination,
+      sharedData,
+    );
+    sharedData = consensusUpdateResult.sharedData;
+
+    return {
+      protocolState: states.waitForPostFundSetup({
+        ...protocolState,
+        postFundSetupState: advanceChannelResult.protocolState,
+        consensusUpdateState: consensusUpdateResult.protocolState,
+      }),
+      sharedData,
+    };
+  }
+
+  return { protocolState, sharedData };
+}
+
+// TODO: This should be an advance channel helper
+function channelSpecificArgs(
+  allocation: string[],
+  destination: string[],
+): { allocation: string[]; destination: string[]; appAttributes: string } {
+  return {
+    allocation,
+    destination,
+    appAttributes: bytesFromAppAttributes({
+      proposedAllocation: allocation,
+      proposedDestination: destination,
+      furtherVotesRequired: 0,
```

Comment:
I think this should be the number of participants, rather than `0`.

Reply:
I think you're right. This should probably be changed in `virtualFunding` as well (where I copied it from)